regrid/regrid.py

```diff
@@ -0,0 +1,251 @@
+            if len(ds_in[var].coords) > 2:
+                coords = ds_in[var].coords.to_index()
+
+                # choose regridding type
+                if coords.names[0] != "time":
+                    raise ValueError("First coordinate should be time")
+                else:
+                    variable_map = {
+                        "SSU": ("SSV", "U"),
+                        "SSV": (None, "skip"),
+                        "uo": ("vo", "U"),
+                        "vo": (None, "skip"),
+                        "taux": ("tauy", "U"),
+                        "tauy": (None, "skip"),
+                    }
+                    if var in variable_map.keys():
+                        var2, pos = variable_map[var]
+                    else:
+                        var2, pos = (None, "T")
+
+                # 3-dimensional data
+                is_3d = False
+                if coords.names[1] == "z_l" or coords.names[1] == "zl":
+                    dims = ["time", "lev", "lat", "lon"]
+                    is_3d = True
+                else:
+                    dims = ["time", "lat", "lon"]
+
+                if pos == "T":
+                    interp_out = self.rg_tt(ds_in[var].values)
+                    da_out = xr.DataArray(interp_out, dims=dims)
+                    da_out.attrs["long_name"] = ds_in[var].long_name
+                    da_out.attrs["units"] = ds_in[var].units
+                    ds_out.append(da_out.to_dataset(name=var))
+
+                if pos == "U":
+                    # interplate u and v to t-point, then rotate currents/winds to
+                    # earth relative before interpolation
+
+                    # interpolate to t-points
+                    interp_u = self.rg_ut(ds_in[var].values)
+                    interp_v = self.rg_ut(ds_in[var2].values)
+
+                    # rotate to earth-relative
+                    if is_3d:
+                        urot = np.zeros(np.shape(interp_u)[1:])
+                        vrot = np.zeros(np.shape(interp_u)[1:])
+                        for k in range(np.shape(interp_u)[1]):
+                            urot[k] = (
+                                interp_u[0, k, :, :] * self.ds_rot.cos_rot
+                                + interp_v[0, k, :, :] * self.ds_rot.sin_rot
+                            )
+                            vrot[k] = (
+                                interp_v[0, k, :, :] * self.ds_rot.cos_rot
+                                - interp_u[0, k, :, :] * self.ds_rot.sin_rot
+                            )
+                    else:
+                        urot = (
+                            interp_u[0, :, :] * self.ds_rot.cos_rot
+                            + interp_v[0, :, :] * self.ds_rot.sin_rot
+                        )
+                        vrot = (
+                            interp_v[0, :, :] * self.ds_rot.cos_rot
+                            - interp_u[0, :, :] * self.ds_rot.sin_rot
+                        )

```

Comment:
I'd like to suggest that we stick to xarray here. That might make it easier to use dask or cupy in the future. Since all of the C grid specific dimensions have been renamed to lon/lat, I don't think we even need the `is_3d` if statement here. I think we can just multiply `interp_u * ds_rot.cos_rot`, and xarray will figure it out based on the coordinate names.
```suggestion
                    # rotate to earth-relative
                    urot = interp_u.isel(time=0) * self.ds_rot.cos_rot + interp_v.isel(time=0) * self.ds_rot.sin_rot
                    vrot = interp_v.isel(time=0) * self.ds_rot.cos_rot - interp_u.isel(time=0) * self.ds_rot.sin_rot
```

Would you be able to test if that works?

It's also not clear to me why the time dimension should be here if we're dropping it anyway, but I'll make a comment about that above.

Reply:
Seems to work!